src/common/tracer.h

```diff
@@ -1,32 +1,101 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
-#ifndef TRACER_H_
-#define TRACER_H_
+#ifndef TRACER_H
+#define TRACER_H
+
+#ifdef HAVE_JAEGER
 
 #define SIGNED_RIGHT_SHIFT_IS 1
 #define ARITHMETIC_RIGHT_SHIFT 1
-
 #include <jaegertracing/Tracer.h>
 
 typedef std::unique_ptr<opentracing::Span> jspan;
 
-namespace jaeger_tracing {
 
-  extern std::shared_ptr<opentracing::v3::Tracer> tracer;
+#else  // !HAVE_JAEGER
+class Value {
+ public:
+  template <typename T> Value(T val) {}
+};
+
+struct span_stub {
+  template <typename T>
+  void SetTag(std::string_view key, const T& value) const noexcept {}
+  void Log(std::initializer_list<std::pair<std::string_view, Value>> fields) {}
+};
+
+class jspan {
+  span_stub span;
+ public:
+  span_stub& operator*() { return span; }
+  const span_stub& operator*() const { return span; }
+
+  span_stub* operator->() { return &span; }
+  const span_stub* operator->() const { return &span; }
+
+  operator bool() const { return false; }
+};
+
+#endif // !HAVE_JAEGER
+
+namespace tracing {
+
+const auto OP = "op";
+const auto BUCKET_NAME = "bucket_name";
+const auto USER_ID = "user_id";
+const auto OBJECT_NAME = "object_name";
+const auto RETURN = "return";
+const auto UPLOAD_ID = "upload_id";
+const auto TYPE = "type";
+const auto REQUEST = "request";
+
+#ifdef HAVE_JAEGER
+
+class Tracer {
+ private:
+  const static std::shared_ptr<opentracing::Tracer> noop_tracer;
+  std::shared_ptr<opentracing::Tracer> open_tracer;
 
-  void init_tracer(const char* tracer_name);
+ public:
+  Tracer(jaegertracing::Config& conf);
```

Comment:
would be greate if you can add a short note with documentation, on how are we supplying config options!

Reply:
I think it would be a mess to describe it.

so I added a c'tor with `service_name` param, so if one would like to initialize a tracer, it could be done easily.
see 2fd4872